src/services/expenses.service.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const { generateId } = require('../helpers/generateId');
+
+let expenses = [];
+const REQUIRE_KEYS = {
+  userId: 0,
+  spentAt: 'string',
+  title: 'string',
+  amount: 0,
+  category: 'string',
+  note: 'string',
+};
+
+const hasRequiredProps = (expense, res) => {
+  for (const prop in expense) {
+    if (!expense.hasOwnProperty(prop)) {
+      res.send('Your data is empty. Please, specify all required fields');
+
+      return false;
+    }
+  }
+
+  if (
+    typeof expense === 'object'
+    && !Array.isArray(expense)
+    && expense !== null
+  ) {
+    for (const key in REQUIRE_KEYS) {
```

Comment:
Try to avoid such nesting, divide logic into separate functions 
![image](https://github.com/mate-academy/node_accounting-app-v2/assets/70009205/c913bd5b-efce-42fd-a36b-f377eb98ba64)